staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1368,22 +1373,46 @@ func (c *cacheWatcher) processEvents(ctx context.Context, initEvents []*watchCac
 	// consider increase size of result buffer in those cases.
 	const initProcessThreshold = 500 * time.Millisecond
 	startTime := time.Now()
-	for _, event := range initEvents {
+
+	initEventCount := 0
+	for {
+		event, err := cacheInterval.Next()
+		if err != nil {
+			defer close(c.result)
+			defer c.Stop()
+
+			klog.V(2).Info(err)
+			errEvent := &watch.Event{
```

Comment:
I'm sorry for changing my mind here, but after thinking more about this, I think this isn't the best option.

We should try to make this change more backward-compatible.
What was happening before (not for the `init events` but still) was that we were simply closing the watch:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L1224-L1231

After thinking more about this, I think we should basically do exactly the same here (which would effectively translate to calling c.Stop()) and not send this event.

It seems that some of the code you wrote here will no longer be needed - sorry for that. But I think we should switch back as this is safer for clients.

[Yes - we won't send "err too old" immediately, but also, next watch will hit it, which is good enough. And more backward-compatible.]

Reply:
That's a good point, I'll make the changes. Thanks!